### PR TITLE
refactor(emqx_resource): Decorate RPCs

### PR DESCRIPTION
**Pull request by k32**

### CODE REVIEW

File: apps/emqx_resource/src/proto/emqx_resource_proto_v1.erl [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+-spec recreate( emqx_resource:instance_id()
+              , emqx_resource:resource_type()
+              , emqx_resource:resource_config()
+              , emqx_resource:create_opts()
+              ) ->
+          {ok, _, emqx_resource:resource_data()} | {error, _Reason}.
+recreate(InstId, ResourceType, Config, Opts) ->
+    emqx_cluster_rpc:multicall(emqx_resource, recreate_local, [InstId, ResourceType, Config, Opts]).
+
+-spec remove(emqx_resource:instance_id()) -> {ok, _, ok} | {error, _Reason}.
```
Comment (thalesmg) [inline apps/emqx_resource/src/proto/emqx_resource_proto_v1.erl:57]: Those `emqx_cluster_rpc:multicall` apparently can return `{retry, _, _, _}` as well.

File: apps/emqx/test/emqx_bpapi_static_checks.erl [hunk 2]
```diff
@@ -49,13 +49,17 @@
 %% List of known RPC backend modules:
 -define(RPC_MODULES, "gen_rpc, erpc, rpc, emqx_rpc").
 %% List of known functions also known to do RPC:
--define(RPC_FUNCTIONS, "emqx_cluster_rpc:multicall/3, emqx_cluster_rpc:multicall/5").
+-define(RPC_FUNCTIONS, "emqx_cluster_rpc:multicall/3, emqx_cluster_rpc:multicall/5, "
+                       "emqx_plugin_libs_rule:cluster_call/3").
```
Comment (thalesmg) [inline apps/emqx/test/emqx_bpapi_static_checks.erl:53]: Q: is this from other branch? Appears to be unrelated to `emqx_resource`.

File: apps/emqx/test/emqx_bpapi_static_checks.erl [hunk 3]
```diff
@@ -49,13 +49,17 @@
...
 %% List of functions in the RPC backend modules that we can ignore:
--define(IGNORED_RPC_CALLS, "gen_rpc:nodes/0, emqx_rpc:unwrap_erpc/1").
+-define(IGNORED_RPC_CALLS, "gen_rpc:nodes/0, emqx_rpc:unwrap_erpc/1, rpc:pmap/3"). % TODO: handle pmap
```
Comment (thalesmg) [inline apps/emqx/test/emqx_bpapi_static_checks.erl:55]: Q: is pmap used here? I only found it in `apps/emqx_connector/src/emqx_connector_mongo.erl`. :thinking:

**Review by thalesmg — APPROVED** [review]

File: apps/emqx/test/emqx_bpapi_static_checks.erl (hunk 2 above)
Comment (k32) [reply apps/emqx/test/emqx_bpapi_static_checks.erl:53]: Correct, but I packed it into the same PR, since it appears to be a minor change. It's in a separate commit, though.

File: apps/emqx/test/emqx_bpapi_static_checks.erl (hunk 3 above)
Comment (k32) [reply apps/emqx/test/emqx_bpapi_static_checks.erl:55]: Yes, it's only used in one place. Once again, packed it into a different PR, but in a separate commit

File: apps/emqx_resource/src/proto/emqx_resource_proto_v1.erl (hunk 1 above)
Comment (k32) [reply apps/emqx_resource/src/proto/emqx_resource_proto_v1.erl]: Fixed.

**Review by thalesmg — APPROVED** [review]

File: apps/emqx_resource/src/emqx_resource.erl [hunk 4]
```diff
@@ -146,7 +146,7 @@ create(InstId, ResourceType, Config) ->
 -spec create(instance_id(), resource_type(), resource_config(), create_opts()) ->
```
Comment (qzhuyan) [inline apps/emqx_resource/src/emqx_resource.erl:146]: Do we need to update the return val spec here since it could return `{retry, _, _,_}` ?

Comment (k32) [reply apps/emqx_resource/src/emqx_resource.erl:146]: This function should handle retry perhaps?

Comment (qzhuyan) [reply apps/emqx_resource/src/emqx_resource.erl:146]: @zhongwencool WDYT?

**Review by qzhuyan — APPROVED** [review]
